src/helper.js

```diff
@@ -314,8 +315,19 @@ function parseHostmask(hostmask) {
 
 function compareHostmask(a, b) {
 	return (
-		(a.nick.toLowerCase() === b.nick.toLowerCase() || a.nick === "*") &&
-		(a.ident.toLowerCase() === b.ident.toLowerCase() || a.ident === "*") &&
-		(a.hostname.toLowerCase() === b.hostname.toLowerCase() || a.hostname === "*")
+		compareWithWildcard(a.nick, b.nick) &&
+		compareWithWildcard(a.ident, b.ident) &&
+		compareWithWildcard(a.hostname, b.hostname)
 	);
 }
+
+function compareWithWildcard(a, b) {
+	a = a.toLowerCase();
+	b = b.toLowerCase();
+	const wildcars_split = a.split("*").map((sub) => {
```

Comment:
Looks like this doesn't support escaping. The horse docs say `*` and `?` should be escapable with a `\`.

But TBH I'm not sure how important of a feature it is to support escaping so I wouldn't be too distraught if you just don't want to. Seems like it shouldn't be too difficult to support though.

Reply:
Yeah... thought about it but then dismissed it as being probably overkill... after all who uses a wildcard in a nick / ident / host.

But while we are at it we might just do it properly don't we?

Reply:
FTR: the spec is very much unclear how you'd then escape the escape char.
Considering that nicks can contain the \ char but not the wildcards we refrain from implementing the escaping.